pkg/controller/core/workload_controller.go

```diff
@@ -276,6 +276,15 @@ func (r *WorkloadReconciler) Create(e event.CreateEvent) bool {
 	r.adjustResources(log, wlCopy)
 
 	if !workload.HasQuotaReservation(wl) {
+		// The scheduler can kick in before the first reconcile when the list of checks is out of sync
```

Comment:
It could be tricky, at the scheduler level,  to make the distinction between a workload that has no checks and one that did go in the first reconcile yet .

We could try to get the number of admission checks of the target queue from the snapshot and don't set the Admission at the same time time if the numbers are different.

Reply:
That sounds reasonable.

Reply:
Or syncing the checks from the clusterqueue in the snapshot.

Reply:
+1 to `Or syncing the checks from the clusterqueue in the snapshot.`.

Reply:
Reworked. Now it checks that the queue checks (retrieved from the snapshot) are all present in the workload list. If not we only do the quota reservation, the workload will transition to fully admitted when the checks are in sync and ready.